suite-native/module-send-receive/src/components/SendReceiveBottomSheet.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { BottomSheet } from '@suite-native/atoms';
+import { AccountsRootState, selectAccountByKey } from '@suite-common/wallet-core';
+
+import { AccountActionStep } from './AccountActionStep';
+import { AccountSelectionStep } from './AccountSelectionStep';
+import { sendReceiveContentType, SendReceiveContentType } from '../contentType';
+import { AddressGenerationStep } from './AddressGenerationStep';
+import { AddressConfirmationStep } from './AddressConfirmationStep';
+import { FreshAddressStep } from './FreshAddressStep';
+
+type SendReceiveBottomSheetProps = {
+    isVisible: boolean;
+    onVisibilityChange: (visible: boolean) => void;
+};
+
+const DEFAULT_CONTENT_TYPE = sendReceiveContentType.chooseAction;
+
+export const SendReceiveBottomSheet = ({
+    isVisible,
+    onVisibilityChange,
+}: SendReceiveBottomSheetProps) => {
+    const [selectedAccountKey, setSelectedAccountKey] = useState<string>('');
+    const [contentType, setContentType] = useState<SendReceiveContentType>(DEFAULT_CONTENT_TYPE);
+
+    const account = useSelector((state: AccountsRootState) =>
+        selectAccountByKey(state, selectedAccountKey),
+    );
+
+    const handleChangeContentType = (type: SendReceiveContentType) => {
+        setContentType(type);
+    };
+
+    const handleSelectAccount = (accountKey: string) => {
+        setSelectedAccountKey(accountKey);
+    };
+
+    const handleClose = () => {
+        onVisibilityChange(false);
+        setSelectedAccountKey('');
+        setContentType(DEFAULT_CONTENT_TYPE);
+    };
+
+    const getSendReceiveContentTitle = () => {
+        switch (contentType) {
+            case sendReceiveContentType.selectAccountToReceive:
+            case sendReceiveContentType.createNewAddressToReceive:
+            case sendReceiveContentType.confirmNewAddressToReceive:
+            case sendReceiveContentType.generatedAddressToReceive: {
+                return 'Receive';
+            }
+            case sendReceiveContentType.selectAccountToSend: {
+                return 'Send';
+            }
+            default:
+        }
+    };
+
+    const getSendReceiveContentComponent = () => {
+        switch (contentType) {
+            case sendReceiveContentType.chooseAction: {
+                return <AccountActionStep onChangeContentType={handleChangeContentType} />;
+            }
+            case sendReceiveContentType.selectAccountToReceive: {
+                return (
+                    <AccountSelectionStep
+                        onChangeContentType={handleChangeContentType}
+                        onSelectAccount={handleSelectAccount}
+                    />
+                );
+            }
+            case sendReceiveContentType.createNewAddressToReceive: {
+                return account ? (
+                    <AddressGenerationStep
+                        account={account}
```

Comment:
I am not sure about passing whole account to all this steps, passing only accountKey and using selector in components itself. We should favor this solution instead of top - down approach which produces lot of uneccessary rerender and data passing all around in props.